src/main/java/com/google/flourbot/entity/action/sheet/SheetAction.java

```diff
@@ -0,0 +1,33 @@
+package com.google.flourbot.entity.action.sheet;
+
+import com.google.flourbot.entity.action.Action;
+import com.google.flourbot.entity.action.ActionType;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public abstract class SheetAction implements Action {
+
+  protected final String sheetUrl;
+
+  public SheetAction( String sheetUrl) {
+    this.sheetUrl = sheetUrl;
+  }
+
+  public abstract ActionType getActionType();
+
+  public String getSheetUrl() {
+    return sheetUrl;
+  }
+
+  public String getSheetId() {
+    String sheetId = "";
```

Comment:
Is there a way we can throw an exception instead of returning an empty sheedId here?

Reply:
I like that you're thinking about ways to make this more robust. getSheetId() will always work if the URL is properly supplied. So maybe it would be more appropriate to do URL checking when the user first makes their bot.

Alternatively, we could catch a bad URL later on when we handle the case where the bot doesn't have permission to write to the desired document. -- I think we're doing this soon

Reply:
Ok sounds good, I didn't now the getSheetId always works with a proper URL